docs/current/sdk/go/959738-get-started.md

```diff
@@ -75,356 +80,440 @@ func build(repoUrl string) error {
   if err != nil {
     return err
   }
-  // <-- New
 
-  // 4. Write the build output to the host
-  // -->
+  // create build/ directory on host
+  outpath := filepath.Join(".", path)
+  err = os.MkdirAll(outpath, os.ModePerm)
+  if err != nil {
+    return err
+  }
+
+  // get reference to current working directory on the host
+  workdir := client.Host().Workdir()
+
+  // write contents of container build/ directory
+  // to the host working directory
   _, err = workdir.Write(ctx, output, dagger.HostDirectoryWriteOpts{Path: path})
+  // highlight-end
   if err != nil {
     return err
   }
-  // <-- New
 
   return nil
 }
 ```
 
-With this new code, the tool will now write the build artifact to the host after the build is complete.
+The revised `build()` function is the main workhorse here, so let's step through it in detail.
 
-- using the Dagger Go SDK, a reference to the host workdir is created with `.Host().Workdir()`
-- in native Go, create a directory where the build artifact will be output
-- create a reference to the build output in the Dagger engine with `Container.Directory().ID()`
-- write the directory to the host with `HostDirectory.Write()`
+- It begins by creating a Dagger client with `dagger.Connect()`, as before.
+- It uses the client's `Git()` function to obtain a reference to the target repository. This function returns a `GitRepository` struct. The struct's `Branch()` function provides details on a specific branch (here, the `main` branch), the `Tree()` function returns the filesystem of the branch, and the `ID()` function returns a reference for the filesystem. This reference is stored in the `src` variable.
+- It initializes a new container from a base image with the `Container().From()` function and returns a new `Container` struct. In this case, the base image is the `golang:latest` image.
+- It mounts the filesystem of the repository branch in the container using the `WithMountedDirectory()` function of the `Container`.
+  - The first argument to this function is the target path in the container (here, `/src`).
+  - The second argument is the directory to be mounted (here, the reference previously created in the `src` variable).
+  It also changes the current working directory to the `/src` path of the container using the `WithWorkdir()` function and returns a revised `Container` with the results of these operations.
+- It uses the `Exec()` function to execute a command in the container - in this case, the command `go build -o PATH`, where `PATH` refers to the `build/` directory in the container. The command and arguments are specified as a Go struct. The `Exec()` function returns a revised `Container` containing the results of command execution.
+- It obtains a reference to the `build/` directory in the container with the `Directory().ID()` function.
+- It copies the build result from the container to the host as follows:
+  - Using native Go, it creates a directory on the host to store the final build output.
```

Comment:
```suggestion
  - Using the Go standard library, it creates a directory on the host to store the final build output.
```
Maybe? "native go" just stuck out to me (in my mind, technically all this code is native go)